src/nupic/algorithms/Connections.cpp

```diff
@@ -102,10 +105,13 @@ Segment Connections::createSegment(UInt32 cell)
   }
   else
   {
+    const UInt32 flatIdx = nextFlatIdx_++;
     segment.idx = cellData.segments.size();
-    cellData.segments.push_back(SegmentData());
-    cellData.segments[segment.idx].flatIdx = nextFlatIdx_++;
+    segment.flatIdx = flatIdx;
```

Comment:
Sort of, but you're right that it deserves a big central description. I added one.